File: backend/app/schemas/organization.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class OrganizationBase(BaseModel):
    name: str = Field(..., max_length=255)
    description: Optional[str] = None
    location: Optional[str] = Field(None, max_length=255)
    website: Optional[str] = Field(None, max_length=255)


class OrganizationCreate(OrganizationBase):
    """Schema for creating a new organization"""

    pass


class OrganizationUpdate(BaseModel):
    """Schema for updating an organization - all fields optional"""

    name: Optional[str] = Field(None, max_length=255)
    description: Optional[str] = None
    location: Optional[str] = Field(None, max_length=255)
    website: Optional[str] = Field(None, max_length=255)


class OrganizationResponse(OrganizationBase):
    """Schema for organization response"""

    id: str
    owner_user_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/types/organization.ts
export interface Organization {
  id: string;
  name: string;
  description: string | null;
  location: string | null;
  website: string | null;
  owner_user_id: number;
  created_at: string;
  updated_at: string;
}

export interface OrganizationCreate {
  name: string;
  description?: string;
  location?: string;
  website?: string;
}

export interface OrganizationUpdate {
  name?: string;
  description?: string;
  location?: string;
  website?: string;
}

export interface OrganizationMember {
  organization_id: string;
  user_id: number;
  role: "admin" | "member";
  added_at: string;
  user_name: string;
  user_email: string;
}
